pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -197,6 +202,15 @@ func (dswp *desiredStateOfWorldPopulator) findAndAddNewPods() {
 		// Keep consistency of adding pod during reconstruction
 		if dswp.hasAddedPods && dswp.podStateProvider.ShouldPodContainersBeTerminating(pod.UID) {
 			// Do not (re)add volumes for pods that can't also be starting containers
+			// However, we do need to potentially remount volumes while terminating -
+			// for example projected volumes
+			uniquePodName := util.GetUniquePodName(pod)
+			var entry = dswp.pods.processedPods[uniquePodName]
+			if time.Since(entry.lastMarkRemount) > time.Duration(10*time.Second) {
+				dswp.actualStateOfWorld.MarkRemountRequired(uniquePodName)
+				entry.lastMarkRemount = time.Now()
+				dswp.pods.processedPods[uniquePodName] = entry
+			}
```

Comment:
making these changes here (marking remount required and directly manipulating processedPods) is _really_ hard to reason about and does not look safe
* markPodProcessed modifies processedPods under lock, this does not
* it is _really_ hard to understand the implications of marking this as remount required here, but then not calling processPodVolumes

I think we should keep the remount marking and processedPods update calls in the same place rather than introducing new places where we update those.

Reply:
Thanks for the feedback. I'm going to try to rework my changes to address your concerns.